tools/autograd/templates/Functions.cpp

```diff
@@ -712,12 +712,17 @@ std::tuple<Tensor, Tensor, Tensor> batchnorm_double_backward(
     const Tensor & ggB,
     const Tensor & gO,
     double eps,
-    const Tensor & save_mean, // not Variable
-    const Tensor & save_std, // not Variable
-    const Tensor & running_mean, // not Variable
-    const Tensor & running_var, // not Variable
+    const Tensor & save_mean_v,
+    const Tensor & save_std_v,
+    const Tensor & running_mean_v,
+    const Tensor & running_var_v,
     bool training) {
 
+  const Tensor& save_mean = static_cast<const Variable&>(save_mean_v).data();
```

Comment:
Are these guaranteed to be defined? Why are we unpacking them at all?

Reply:
The back story is I couldn't figure out if it was a supported operation to directly pass Tensors (not Variables) to an ATen derivative function, so I compromised with turning them all into Variables and unpacking them. `save_mean` and `save_std` used to be possibly undefined but that broke things, so they're now always defined (I need to fix some stuff elsewhere in this patch, as you have noticed).